src/oca_github_bot/tasks/migration_issue_bot.py

```diff
@@ -0,0 +1,103 @@
+                    cwd=clone_dir,
+                )
+                check_call(["git", "checkout", pr_branch], cwd=clone_dir)
+                if not _user_can_merge(
+                    gh, org, repo, username, clone_dir, target_branch
+                ):
+                    github.gh_call(
+                        gh_pr.create_comment,
+                        f"Sorry @{username} you are not allowed to mark the addon to"
+                        f"be migrated.\n\n"
+                        f"To do so you must either have push permissions on "
+                        f"the repository, or be a declared maintainer of all "
+                        f"modified addons.\n\n"
+                        f"If you wish to adopt an addon and become it's "
+                        f"[maintainer]"
+                        f"(https://odoo-community.org/page/maintainer-role), "
+                        f"open a pull request to add "
+                        f"your GitHub login to the `maintainers` key of its "
+                        f"manifest.",
+                    )
+                    return
+            # Assign milestone to PR
+            milestone = _create_or_find_branch_milestone(repo, target_branch)
+            gh_pr.issue().edit(milestone=milestone.number)
+            # Find issue
+            issue_title = f"Migration to version {target_branch}"
+            issue = False
+            for i in repo.issues(milestone=milestone.number):
+                if i.title == issue_title:
+                    issue = i
+                    break
+            if not issue:
+                github.gh_call(
+                    gh_pr.create_comment,
+                    f"There's no issue in this repo with the title '{issue_title}' "
+                    f"and the milestone {target_branch}, so not possible to add "
+                    f"the comment.",
+                )
+                return
+            # Change issue to add the PR in the module list
+            lines = []
+            added = False
+            module_list = False
+            new_line = f"- [ ] {module} - By @{gh_pr.user.login} - #{gh_pr.number}"
+            for line in issue.body.split("\n"):
+                if line.startswith(f"- [ ] {module}"):
+                    lines.append(new_line)
+                    continue
+                elif not added:
+                    splits = re.split(r"- \[ \] ([0-9a-zA-Z_]*)", line)
+                    if len(splits) >= 2:
+                        # Flag for detecting if we have passed already module list
+                        module_list = True
+                        line_module = splits[1]
+                        if line_module > module:
+                            lines.append(new_line)
+                            added = True
+                    elif module_list:
+                        lines.append(new_line)
+                        added = True
+                lines.append(line)
```

Comment:
This block that updates the body text could be easily extracted in a standalone function with unit tests.

Reply:
@pedrobaeza I'm not asking for a complete integration test. I'm just asking to extract this inner part that does the updating in a function with a couple of unit tests.

Reply:
OK, I will check.